3 - PB/Documentazione interna/Norme di Progetto/Norme di Progetto.typ

```diff
@@ -1697,43 +1697,74 @@ Gli argomenti principali trattati nel documento sono due:
+
+In caso di modifica è necessario reiterare l'esecuzione dei test sviluppati, al fine di garantire che il software sviluppato e testato in precedenza funzioni ancora come previsto. In caso contrario, si parlerebbe di regressione.
+
+Considerando la complessità degli elementi grafici e delle interazioni utente che non possono essere facilmente o economicamente testati in modo automatico, viene istituita una modalità di testing manuale. In questa modalità, è responsabilità dello sviluppatore dell'elemento di sistema (Programmatore) verificare che il codice scritto funzioni correttamente. Questo vale sia per il codice sviluppato in proprio sia per quello riutilizzato o esterno, che viene adattato per soddisfare i requisiti specifici.
+
+Per aiutare il lavoro di verifica da parte del Verificatore, vengono riportate in pr tutte le funzionalità che si ha necessità di controllare manualmente.
+
+==== Implementazione
+
+Decisa una strategia di testing per l'elemento di sistema e scritti i relativi test di unità, il Programmatore può quindi procedere allo sviluppo o all'adeguamento del software.
+
+Per garantire una standardizzazione e una migliore gestione del codice software, sono state implementate diverse pratiche di organizzazione e normativa come segue:
+
++ approccio organizzato alla struttura delle cartelle:
+  - ciascuna categoria di elementi è assegnata a una cartella specifica (@repository-github):
+    - tutti i file correlati ad uno specifico pattern sono organizzati all'interno di una cartella dedicata esclusivamente a quel pattern;
+    - tutti i file riguardanti l'interfaccia utente sono raggruppati in una cartella apposita;
+    - all'interno di alcune cartelle, come "ServerActions", è presente una ulteriore suddivisione degli elementi in sotto-cartelle per zone specifiche, come Zone, Bin, ecc., seguendo una logica di organizzazione interna coerente.
+
++ adottato il principio per cui un elemento/componente/classe = un file. Pratica che semplifica la gestione e la ricerca dei componenti all'interno del progetto, garantendo una maggiore chiarezza e coerenza nel codice.
 
-Data la mole di elementi grafici o interazioni utente che non sono né facili, né economici da testare automaticamente viene definita una modalità di testing manuale: chi sviluppa l'elemento di sistema è responsabile della verifica del corretto funzionamento del codice scritto. Questo vale anche per codice riutilizzato o codice esterno che viene adattato per soddisfare i requisiti richiesti.
++ nomenclatura delle variabili, dei metodi e delle funzioni, si attiene al principio che i nomi devono essere descrittivi e parlanti. Questo significa che i nomi dovrebbero riflettere chiaramente il loro scopo e la loro funzione all'interno del sistema, facilitando così la comprensione e la manutenzione del codice.
```

Comment:
Riformulerei in:

"i nomi di variabili, metodi e funzioni devono essere parlanti, ovvero devono riflettere il loro scopo e la loro funzione all'interno del sistema, così da facilitare la comprensione e manutenzione del codice;"